sdk/servicebus/service-bus/src/receivers/receiverCommon.ts

```diff
@@ -245,23 +246,25 @@ export function settleMessageOperation(
     throw error;
   }
 
-  // Message Settlement with managementLink
-  // 1. If the received message is deferred as such messages can only be settled using managementLink
-  // 2. If the associated receiver link is not available. This does not apply to messages from sessions as we need a lock on the session to do so.
-  if (isDeferredMessage || ((!receiver || !receiver.isOpen()) && !isDefined(message.sessionId))) {
-    return context
-      .getManagementClient(entityPath)
-      .updateDispositionStatus(message.lockToken!, operation, {
-        ...options,
-        associatedLinkName,
-        sessionId: message.sessionId
-      })
-      .catch((err) => {
-        throw translateServiceBusError(err);
-      });
-  }
+  try {
+    // Message Settlement with managementLink
+    // 1. If the received message is deferred as such messages can only be settled using managementLink
+    // 2. If the associated receiver link is not available. This does not apply to messages from sessions as we need a lock on the session to do so.
+    if (isDeferredMessage || ((!receiver || !receiver.isOpen()) && !isDefined(message.sessionId))) {
+      return context
+        .getManagementClient(entityPath)
+        .updateDispositionStatus(message.lockToken!, operation, {
+          ...options,
+          associatedLinkName,
+          sessionId: message.sessionId
+        });
+    }
 
-  return receiver!.settleMessage(message, operation, options).catch((err) => {
+    await receiver!.settleMessage(message, operation, options);
+    // delay (setTimeout) ensures that the delivery is popped, size is decremented with respect to the settlement that was done
+    await delay(0);
```

Comment:
So what happens if we don't have this delay? Wondering if this is just for our tests or if it actually causes bad behavior if we omit this.

Reply:
**It is bad behavior if we omit this.**

`receiver.settlementNotifierForSubscribe?.();` calls the creditManager.postProcessing() which adds credits if there are empty slots present.
Empty slots are supposed to be alloted because a message has been settled and the delivery is removed from the buffer.

### Workflow
1. settle one message 
2. settling a message triggers the post process script to add more credits
    - before adding more credits, it checks for the number of empty slots (capacity - size)
3. when a message is settled, rhea removes it from the buffer, size is decremented, (which should create an empty slot at step 2)

In node, step 3 happens and then step 2
In browser, step 2 happens and then step 3